IntroToCSharp/Shared/Components/Page/PageRegistry.cs

```diff
@@ -0,0 +1,135 @@
+    {
+        int order = 0;
+        AddSection("Getting Started", order++);
+        AddSection("Activities", order++);
+        AddSection("Lessons", order++);
+        AddSection("Variables", "Lessons", order++);
+        AddSection("Challenges", order++);
+        AddSection("Journal", order++);
+        AddSection("Social", order++);
+    }
+
+    private PageRegistry Init()
+    {
+        var type = typeof(Page);
+        var pages = AppDomain.CurrentDomain.GetAssemblies()
+            .SelectMany(s => s.GetTypes())
+            .Where(p => type.IsAssignableFrom(p));
+        foreach(var page in pages)
+        {
+            FieldInfo? field = page.GetField("PageRef");
+            if (field == null) continue;
+            PageRef? pageRef = (PageRef?)field.GetValue(null);
+            if (pageRef == null) continue;
+            this.AddPage((PageRef)pageRef);
+        }
+        return this;
+    }
+
+    /// <summary>
+    /// Adds the specified section to the registry specifying its
+    /// order in the list.
+    /// </summary>
+    private MenuItem AddSection(string name, int order)
+    {
+        if (!_items.TryGetValue(name, out MenuItem? item) || item == null)
+        {
+            item = new MenuItem(name, null, order, true);
+            _items[name] = item;
+        }
+        else
+        {
+            item.Order = order;
+        }
+        return item;
+    }
+
+    /// <summary>
+    /// Adds the specified subsection to the registry specifying its
+    /// order in the parent section.
+    /// </summary>
+    private MenuItem AddSection(string name, string parent, int order)
+    {
+        MenuItem item = AddSection(name, order);
+        MenuItem parentItem = GetParent(parent);
+        item.Parent = parentItem;
+        parentItem.Children.Add(item);
+        return item;
+    }
+
+    private MenuItem GetParent(string parent)
+    {
+        if (!_items.TryGetValue(parent, out MenuItem? parentItem) || parentItem == null)
+        {
+            parentItem = new MenuItem(parent, null, -1, true);
+            _items[parent] = parentItem;
+        }
+        return parentItem;
+    }
+
+    private void AddPage(PageRef p)
+    {
+        if (_items.ContainsKey(p.Name)) throw new InvalidOperationException($"The specified page {p.Name} already exists.");
+        MenuItem parent = GetParent(p.Parent);
+        MenuItem page = new (p.Name, p.Href, p.Order, p.IsAdventure, parent);
+        page.Parent = parent;
+        parent.Children.Add(page);
+        _items[p.Name] = page;
+    }
+
+    public IEnumerable<MenuItem> RootElements => _items.Values.Where(item => item.Parent == null).OrderBy(item => item.Order);
+
+}
+
+/// <summary>
+/// A MenuItem represents the data necessary for an item to 
+/// be rendered on the NavMenu.
+/// </summary>
+public class MenuItem : IComparable<MenuItem>
+{
+    private readonly string _name;
+    private readonly string? _href;
+    private readonly bool _isAdventure;
```

Comment:
Should _isAdventure be true until it is set to false?

Reply:
It is a required parameter of the constructor so there is no default value.